task/run-opm-command-oci-ta/0.1/run-opm-command-oci-ta.yaml

```diff
@@ -0,0 +1,266 @@
+        runAsUser: 0
+      when:
+        - input: "$(params.IDMS_PATH)"
+          operator: notin
+          values: ["", "null"]
+      command: ["bash", "-c"]
+      args:
+        - |
+          #!/bin/bash
+          set -euo pipefail
+          # shellcheck source=/dev/null
+          source /utils.sh
+
+          # --- Validate input arguments ---
+          if [[ $# -ne 2 ]]; then
+              echo "Error: The script requires 2 arguments: <path_to_idms_file> <path_to_catalog_file>" >&2
+              exit 1
+          fi
+
+          IDMS_FILE=$1
+          CATALOG_FILE=$2
+
+          # --- Check file existence ---
+          if [[ ! -f "$CATALOG_FILE" ]]; then
+              echo "Error: Catalog file not found at: ${CATALOG_FILE}" >&2
+              exit 1
+          fi
+
+          # --- Create temporary files ---
+          # Using mktemp for safe temporary file creation
+          IMDS_MAP_TMP_FILE="$(mktemp)"
+          INPUT_CATALOG_FILE="$(mktemp)"
+          CATALOG_OUTPUT_TMP_FILE="$(mktemp)"
+
+          # Register a trap function to clean up temporary files on script exit (even on error)
+          trap 'rm -f "$IMDS_MAP_TMP_FILE" "$INPUT_CATALOG_FILE" "$CATALOG_OUTPUT_TMP_FILE"' EXIT
+
+          # Create a working copy of the catalog to avoid modifying the original while reading
+          cp "$CATALOG_FILE" "$INPUT_CATALOG_FILE"
+
+          # --- Process mirror mapping ---
+          if [[ -f "${IDMS_FILE}" ]]; then
+            echo "Processing ImageDigestMirrorSet from ${IDMS_FILE}..."
+            mirror_set_yaml=$(cat "${IDMS_FILE}")
+            process_image_digest_mirror_set "${mirror_set_yaml}" > "$IMDS_MAP_TMP_FILE"
+
+            # KEY CHECK: Verify that the generated file is valid JSON and not empty.
+            if ! jq -e . "$IMDS_MAP_TMP_FILE" > /dev/null 2>&1; then
+              echo "Error: The 'process_image_digest_mirror_set' function did not generate valid JSON." >&2
+              echo "Content of the generated file:" >&2
+              cat "$IMDS_MAP_TMP_FILE" >&2
+              exit 1 # Exiting the script with an error
+            fi
+            echo "ImageDigestMirrorSet was processed successfully."
+          else
+            echo "ImageDigestMirrorSet file not found at: ${IDMS_FILE}."
+            echo "Skipping replacement."
+            exit 0
+          fi
+
+          MAPPING_JSON_CONTENT=$(cat "$IMDS_MAP_TMP_FILE")
+
+          echo "Replacing values in relatedImages..."
+          jq --argjson mapping "$MAPPING_JSON_CONTENT" '
+            # Prepare the mapping data
+            ($mapping // {}) as $safe_mapping |
+            # Sort sources from longest to shortest to avoid incorrect replacement on nested paths
+            # (e.g., "registry/org/app" is processed before "registry/org")
+            ($safe_mapping | keys_unsorted | sort_by(length) | reverse) as $sorted_sources |
+
+            # Function to replace the URL in a single image object
+            def replace_image_url:
```

Comment:
Can't we use `https://github.com/konflux-ci/konflux-test/blob/main/test/utils.sh#L838` instead to replace pullspecs?

Reply:
Moved logic to bash and koflux-test image. 
Please review.